### ISSUE TITLE
Review abnormalCellProliferation

### ISSUE THREAD

**Issue by srobb1** [obophenotype/upheno#316]
[abnormalCellProliferation.yaml](https://github.com/obophenotype/upheno/blob/master/src/patterns/abnormalCellProliferation.yaml)

Currently PLANP would be unable to use this pattern since it is restricted to 

anatomical entity: UBERON:0001062 

The planarian anatomy ontology ([PLANA](https://www.ebi.ac.uk/ols/ontologies/plana/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPLANA_0000400)) has 'anatomical entity' but it has a PLANA id and has UBERON:0001062 as a dbxref.

There are quite a few other anatomy ontologies that have the same design.

I suggest using
entity: BFO:0000001

**Comment by matentzn** [comment]
In order for anatomy patterns to interoperate, we need to align the external anatomy ontologies (such as wbbt, zfa, plana, xao) with UBERON. This is going to be a gradual process we will realise later in the process by curating xrefs between those ontologies and Uberon. 

So with that goal in mind it is best to leave all the UBERON restrictions in there; you can use the pattern even without uberon alignment in any case; problems occur only when checking pattern conformance later.

**Comment by srobb1** [comment]
Great! That will be awesome for the xrefs to be used to link ontologies!!

**Comment by balhoff** [comment]
`inheres in part of is just a chain of something that inheres in something that is part of something else. So if you had a triangular that inheres in a dorsal fin, and dorsal fin is part of body, you could say ‘triangular and inheres in part of some body’. But it doesn’t inhere in it; the body is not triangular.

**Comment by drseb** [comment]
I drew a picture a while ago, maybe this also helps you:
<img width="666" alt="screenshot 2019-02-01 at 20 31 28" src="https://user-images.githubusercontent.com/5172897/52145100-6e414d00-2660-11e9-863b-ae1ba1e11819.png">

**Comment by matentzn** [comment]
@nicolevasilevsky can you give some explanations why? just learning pato :P

**Comment by nicolevasilevsky** [comment]
I am only basing this on the fact that MP uses quality instead of process quality - I actually don't know what is the appropriate term to use.

It doesn't look like MP uses process quality at all in any of the logical defs.

HP uses it in only two terms:

HP_0012647 'Abnormal inflammatory response'
'has part' some 
('process quality' and ('inheres in' some 'inflammatory response') and ('has modifier' some abnormal))

HP_0040224 'Abnormality of fibrinolysis'
'has part' some 
('process quality' and ('inheres in part of' some fibrinolysis) and ('has modifier' some abnormal))

**Comment by matentzn** [comment]
I dont mind either way; my general feeling is to pick the constraints that most accurately reflect the intention of the pattern; if this pattern is about abnormal process quality, it should use that, no matter who is using it how at the moment; lets see what the others say.

**Comment by ybradford** [comment]
I agree that if entity= process a process quality should be used. This is how it was originally intended to be used as I understand it.  This is the way ZFIN has used it and constrains annotations with this in mind.

**Comment by Clare72** [comment]
DPO also uses children of 'process quality' for processes.

For this pattern, I don't think the name and definition will work well if you want to use a cell type e.g. neuroblast, rather than a location e.g. brain. I'm not sure if type and location style terms can be incorporated into the same pattern, so there might need to be another pattern, maybe "abnormalCellProliferationByCellType" and rename the current one "abnormalCellProliferationByLocation".

**Comment by matentzn** [comment]
If I understand @cmungall and his OBO core efforts correctly, we will soon have some kind of minimal biological upper layer which has a concept called 'Anatomical structure', that encompasses both cells and anatomical entities/regions. However, it is indeed debatable whether something that affects all cells of a certain kind is the same as something that effects an anatomical entity; i mean AFAIK cells of one type can be kind of scattered everywhere and not really belong to a coherent anatomical structure. Food for thought! @dosumis @cmungall

**Comment by drseb** [comment]
Wouldn’t it technically be possible to state in the pattern that multiple classes can be possible here? E.g. „cell“ or „anatomical entity“?

Sent from mobile

**Comment by matentzn** [comment]
No, by design DOSDP is taking the **dead simple** very seriously.. You will need to have two patterns.

**Comment by mellybelly** [comment]
a few things that may be relevant. In CARO cells are anatomical entities, I thought this was still true in the OBO Core (as per @matentzn comment above). Agree that process quality should be used here, proliferation is a process. It does seem like we need two patterns, one that is simply abnormal process, and one where an abnormal process is part of another anatomical entity (and thereby infer that there may be phenotypic outcomes upon that entity)? Is this what you mean @matentzn ?

**Comment by matentzn** [comment]
Oh, I might have misread the pattern before. Yeah @mellybelly you are right, probably we need two patterns here:
1. abnormalCellProliferation (or what @Clare72 calls abnormalCellProliferationByCellType)
2. abnormalCellProliferationInALocation

The domain should be CELL TYPE, not anatomical structure! This pattern is AFAICT only about proliferation of cells and not about proliferation of anything else.

**Comment by drseb** [comment]
I thought the problem is the missing class that unites cells and other anatomical entities!?

**Comment by drseb** [comment]
Ok. Understood now. Sorry

**Comment by matentzn** [comment]
Its all still a bit confusing, but soon there will be clarity I think :-)

**Comment by dosumis** [comment]
> equivalentTo:
> text: "'has_part' some ('process quality' and ('inheres_in_part_of' some (cell_proliferation and ('occurs in' some %s))) and ('qualifier' some 'abnormal'))"
> vars:
> - anatomical_entity

We should align compound entities with source ontologies - using the relationships they use.   This maximises inference.

In this case we should use whatever GO uses. GO has:

acinar cell proliferation EquivalentTo:
'cell population proliferation' and (acts_on_population_of some 'acinar cell')

cell proliferation in forebrain:
'cell population proliferation' and ('occurs in' some forebrain)

We should therefore have two patterns reflecting this:

abnormal_proliferation_of
abnormal cell proliferation_in

The range of the var in the first pattern should be 'cell'

Ideally the range of the var in the second pattern would be 'multicellular anatomical structure'. This is in CARO, but we need to check whether Uberon supports it.  Failing that we can use 'anatomical structure' which subsumes cell but name write the pattern description and name the var to make clear we want multicellular structures only.

**Comment by matentzn** [comment]
+1 on Davids suggestion!

**Comment by dosumis** [comment]
Discussion on Phenotype  call:

All agree that we should follow GO and have two patterns as described above - 

X cell proliferation:
'cell population proliferation' and (acts_on_population_of some 'X')

cell proliferation in X:
'cell population proliferation' and ('occurs in' some X)

**Comment by nicolevasilevsky** [comment]
I will create two new patterns for this

**Comment by nicolevasilevsky** [comment]
Here is one:
https://github.com/obophenotype/upheno/blob/master/src/patterns/dosdp-dev/abnormalProliferationOf.yaml

**Comment by nicolevasilevsky** [comment]
I am confused, should abnormal cell proliferation in an anatomical entity have the logical def:
'cell population proliferation' and (acts_on_population_of some 'X') and ('occurs_in some X)

?

**Comment by nicolevasilevsky** [comment]
here is one pattern:
https://github.com/obophenotype/upheno/blob/master/src/patterns/dosdp-dev/abnormalCellProliferationInAnatomicalEntity.yaml

not sure if this is right, but it is a starting place

**Comment by nicolevasilevsky** [comment]
is there an existing relation for acts_on_population_of ?

**Comment by nicolevasilevsky** [comment]
@dosumis and @matentzn

**Comment by dosumis** [comment]
> is there an existing relation for acts_on_population_of ?

There is in GO. I didn't realise until now, but this seems to be a relation never made it out of GO into RO.   

It has a the iri: http://purl.obolibrary.org/obo/GOREL_0001006 

I think the only public release file you'll see this in is go-plus.owl

**Comment by dosumis** [comment]
It's in go-base - so should be usable:

**Comment by matentzn** [comment]
We have
- [X] http://purl.obolibrary.org/obo/upheno/patterns-dev/abnormalCellProliferationInAnatomicalEntity.yaml

We still need 
- [ ] abnormalCellProliferationActingOnCellsOfType.yaml
- [ ] abnormalCellProliferationActingOnCellsOfTypeInLocation.yaml

@nicolevasilevsky would you mind making PRs for those?

**Comment by matentzn** [comment]
@nicolevasilevsky on closer look,

https://github.com/obophenotype/upheno/blob/master/src/patterns/dosdp-dev/abnormalCellProliferationInAnatomicalEntity.yaml is also wrong! It inheres in a GO term! Can you fix? :) Thaaanks (moved it to dosdp workshop for the time being)

**Comment by matentzn** [comment]
Fixed above pattern last comment.
